docs/developer/architecture/centralized-cli.rst

```diff
@@ -0,0 +1,62 @@
+Centralized CLI
+===============
+
+Context
+-------
+
+MetalK8s comes with a set of services to operate and monitor the K8s cluster.
+All operations that need to be performed by the Platform Administrator could be
+categorized as follow:
+
+- Cluster Resources Administration (Nodes, Volumes, Deployments, ...)
+- Cluster Administration (Install, Upgrade, Downgrade, Backup, Restore, ...)
+- Solution Administration (CRUD Environment, Import/Remove Solution, ...)
+- Cluster Service Administration (Configure Dex, Prometheus, Alert Manager,
+  ...)
+
+K8s provides the kubectl CLI, which is distributed and enables all kind of
+interaction with all Kubernetes resources, through apiserver, but its usage
+often requires to build verbose YAML files.
+
+Currently, MetalK8s provides other set of scripts or manual procedures, but
+those are located in various locations, their usage may vary and they are not
+developed using the same logic.
+
+This makes the CLI and associated documentation not super intuitive and it
+also makes the maintenance more expensive in the long term.
+
+The goal of the project is to provide MetalK8s administrator with an intuitive
+and easy to use set of tools in order to administrate and operate a finite set
+of functionalities.
+
+Because kubectl CLI is already in place and is well known by Kubernetes
+administrators, it will be used as a standard to follow for all other MetalK8s
+CLIs:
+
+- CLI follows kubectl style: kubectl <action> <object>
+- CLI provides an exhaustive help, per action, with relevant examples
+- CLI is not interactive (except maybe for very first install script)
+- CLI should not require password input
+- CLI implementation relies on secure APIs
+- CLI support <action> completion for easy discovery
+- CLI output is standardized and human readable by default
+- CLI output can be formatted in JSON or YAML
+- CLI can be executed from outside of the cluster (except maybe for very first
+  install script)
```

Comment:
> except maybe for very first install script

No, not only. You also have for example the iso-manager, upgrade and downgrade script that need to run on the bootstrap node. For me you should remove this requirement or have a better "except" part but ... I don't know how you can define it for me it's something like "except for script that need to (re)start salt-master container" but it maybe not only this

Reply:
@TeddyAndrieux the requirement is to be able to run everything from outside of the cluster, as we can do with kubectl. It is maybe not possible, you will say it in the design.

Reply:
I'm with @TeddyAndrieux here. Putting something in a requirement means "We really need this for some customer use-case". As such, either it's a must (requirement) and we need to figure out some design to accommodate the need, or it's not a requirement.

Now, in the case where 'local things' are required (indeed, e.g. ISO import, where one needs to refer to a *local* path to an ISO), *requiring* this to be available 'from outside' seems odd.

Reply:
Well you can execute remotely some 'local things' with local path, and yes, it's a bit weird.
The point is why we don't want this ? Because of technical difficulties or conceptual things or security things ?
Moreover could the CLI upgrade itself ?

Reply:
@NicolasT @xaf-scality @TeddyAndrieux let's discuss this one during today's meeting.

Reply:
removed, only kubectl metalk8s plugin should be executable from outside of the cluster